.NET/Microsoft.Recognizers.Text.Number/Chinese/Parsers/ChineseNumberParser.cs

```diff
@@ -32,62 +32,48 @@ public override ParseResult Parse(ExtractResult extResultChs)
 
             if (extra.Contains("Per"))
             {
-                ret = PerParseChs(simplifiedExtResultChs);
+                ret = PerParseChs(simplifiedExtResult);
             }
             else if (extra.Contains("Num"))
             {
-                simplifiedExtResultChs.Text = ReplaceFullWithHalf(simplifiedExtResultChs.Text);
-                ret = DigitNumberParse(simplifiedExtResultChs);
-                if (Config.NegativeNumberSignRegex.IsMatch(simplifiedExtResultChs.Text) && (double)ret.Value > 0)
+                simplifiedExtResult.Text = PatternUtil.NormalizeCharWidth(simplifiedExtResult.Text, Culture.Chinese);
+                ret = DigitNumberParse(simplifiedExtResult);
+                if (Config.NegativeNumberSignRegex.IsMatch(simplifiedExtResult.Text) && (double)ret.Value > 0)
                 {
                     ret.Value = -(double)ret.Value;
                 }
                 ret.ResolutionStr = ret.Value.ToString();
             }
             else if (extra.Contains("Pow"))
             {
-                simplifiedExtResultChs.Text = ReplaceFullWithHalf(simplifiedExtResultChs.Text);
-                ret = PowerNumberParse(simplifiedExtResultChs);
+                simplifiedExtResult.Text = PatternUtil.NormalizeCharWidth(simplifiedExtResult.Text, Culture.Chinese);
+                ret = PowerNumberParse(simplifiedExtResult);
                 ret.ResolutionStr = ret.Value.ToString();
             }
             else if (extra.Contains("Frac"))
             {
-                ret = FracParseChs(simplifiedExtResultChs);
+                ret = FracParseChs(simplifiedExtResult);
             }
             else if (extra.Contains("Dou"))
             {
-                ret = DouParseChs(simplifiedExtResultChs);
+                ret = DouParseChs(simplifiedExtResult);

```

Comment:
Ideally methods should not have language prefixes. DouParseChs should not end with Chs.
Also, please rename the methods to be more easily readable. This one could be ParseDouble.